ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,1180 @@
+        const chart = this.chart,
+            drilldownLevels = chart.drilldownLevels;
+
+        // Calculate on which level we are now.
+        this.level = drilldownLevels && drilldownLevels[drilldownLevels.length - 1] &&
+        drilldownLevels[drilldownLevels.length - 1].levelNumber + 1 || 0;
+    }
+
+    /**
+     * This method creates an array of arrays containing a level number
+     * with the corresponding series/point.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#createList
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     * @return {any}
+     *        Formatted text.
+     */
+    public createList(this: Breadcrumbs): any {
+        const breadcrumbsList: Array<Array<number|null|Point|SeriesType|SVGAElement>> = this.breadcrumbsList || [],
+            chart = this.chart,
+            drilldownLevels = chart.drilldownLevels;
+
+        // The list is based on drilldown levels from the chart object
+        if (drilldownLevels && drilldownLevels.length) {
+            // Add the initial series as the first element.
+            if (!breadcrumbsList[0]) {
+                breadcrumbsList.push([0, (drilldownLevels[0].seriesOptions as Point)]);
+            }
+
+            const lastBreadcrumb = breadcrumbsList[breadcrumbsList.length - 1];
+
+            drilldownLevels.forEach(function (level): void {
+                // If level is already added to breadcrumbs list,
+                // don't add it again- drilling categories
+                // + 1 because of the wrong levels numeration
+                // in drilldownLevels array.
+                if (level.levelNumber + 1 > (lastBreadcrumb as any)[0]) {
+                    breadcrumbsList.push([level.levelNumber + 1, (level.pointOptions as Point)]);
+                }
+            });
+        }
+        return breadcrumbsList;
+    }
+
+    /**
+     * Destroy the chosen buttons and separators.
+     *
+     * @requires  modules/breadcrumbs
+     *
+     * @function Highcharts.Breadcrumbs#destroyGroup
+     * @param {Highcharts.Breadcrumbs} this
+     *        Breadcrumbs class.
+     */
+    public destroyGroup(this: Breadcrumbs): void {
+        const breadcrumbs = this,
+            chart = breadcrumbs.chart,
+            breadcrumbsList = breadcrumbs.breadcrumbsList,
+            breadcrumbsOptions = breadcrumbs.options;
+
+        // The full path of buttons is visible.
+        if (breadcrumbsOptions.showFullPath && breadcrumbs.breadcrumbsGroup) {
+            breadcrumbsList.forEach(function (el): void {
+                const button = el[2],
+                    separator = el[3];
+
+                // Remove SVG elements fromt the DOM.
+                button && button.destroy();
+                separator && separator.destroy();
+            });
+            // Clear the breadcrums list array.
+            breadcrumbsList.length = 0;
+            this.breadcrumbsGroup.destroy();
```

Comment:
Q: Do we need to destroy each button first if we destroy the group later?

Reply:
I don't think we need, I have removed it from destroyGroup method.

Reply:
hm, on the other hand, we may need it, lack of it cause an error in one test